cs/v1/CS_v1.0.0.yaml

```diff
@@ -874,6 +960,303 @@ components:
+              maxLength: 65
+            stateRegion:
+              type: string
+              x-stoplight:
+                id: 3k7wb2fib7401
+              example: North Holland
+              maxLength: 65
+            country:
+              type: string
+              x-stoplight:
+                id: enhizhmuskzj6
+              maxLength: 75
+    FacilitySMDGLocation:
+      title: Facility SMDG Location
+      x-stoplight:
+        id: wjfsbdac9a7nz
+      type: object
+      description: An interface used to express a location using a `Facility` by the `SMDG` code list. The `facilitySMDGCode` does not contain the `UNLocationCode` - this should be provided in the `UnLocationCode` attribute.
+      required:
+        - locationType
+        - UNLocationCode
+        - facilitySMDGCode
+      properties:
+        locationName:
+          type: string
+          x-stoplight:
+            id: p82efb2odh99y
+          description: The name of the location.
+          example: Port of Amsterdam
+          maxLength: 100
+        locationType:
+          type: string
+          x-stoplight:
+            id: wvrz1c6visndu
+          example: FACS
+          description: Discriminator used to identify this as a `Facility Location` interface only using `SMDG` code list.
+          maxLength: 4
+        UNLocationCode:
+          type: string
+          x-stoplight:
+            id: 7kpvxrrdljsb5
+          example: NLAMS
+          pattern: '^[A-Z]{2}[A-Z2-9]{3}$'
+          minLength: 5
+          maxLength: 5
+          description: "The UN Location code specifying where the place is located. The pattern used must be\r\n\r\n - 2 characters for the country code using [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)\r\n - 3 characters to code a location within that country. Letters A-Z and numbers from 2-9 can be used\r\n\r\nMore info can be found here: [UN/LOCODE](https://en.wikipedia.org/wiki/UN/LOCODE)"
+        facilitySMDGCode:
+          type: string
+          x-stoplight:
+            id: yndr8amfy18nr
+          example: ACT
+          description: "The code used for identifying the specific facility. This code does not include the UN Location Code.\r\n\r\nThe codeList used by SMDG is the [SMDG Terminal Code List](https://smdg.org/wp-content/uploads/Codelists/Terminals/SMDG-Terminal-Code-List-v20210401.xlsx)"
+          maxLength: 6
+    UNLocationLocation:
+      title: UNLocation Location
+      x-stoplight:
+        id: inxjf3adrpkge
+      type: object
+      description: An interface used to express a location using a `Un Location Code`.
+      additionalProperties: false
+      required:
+        - locationType
+        - UNLocationCode
+      properties:
+        locationName:
+          type: string
+          x-stoplight:
+            id: uwy2d1d9tkdku
+          description: The name of the location.
+          example: Port of Amsterdam
+          pattern: ^\S+(\s+\S+)*$
```

Comment:
This pattern uses the "old" pattern and does not cater for "Catastrophic Backtracking"